From d07290f9430066cded25d61463d8c99962e55fe7 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Mon, 14 Nov 2022 07:19:48 -0800
Subject: [PATCH] Fix `update-chainparams.sh` so it can be ran on MacOS X
 (#294)

Script now looks for an env var called `CONFIG_FILE` that will override
the default config location. Can be ran as:

```
CONFIG_FILE=CONFIG_FILE=~/Library/Application\ Support/Lotus/lotus.conf contrib/source-control-tools/patch-recipes/update-chainparams.sh
```

Rebase provenance: this historical change was adapted onto the pinned
Bitcoin ABC base. Source diff is the recorded Lotus V1 replay commit below.

Historical-source-commit: 6dc14e91fe117c628e380c2d4a5672e3f7d9e04b
Recorded-replay-commit: 209afe3d7343fb56686c5692e90407408940d52f

diff --git a/contrib/source-control-tools/patch-recipes/update-chainparams.sh b/contrib/source-control-tools/patch-recipes/update-chainparams.sh
index a39fcf7f93..885a22f8f2 100755
--- a/contrib/source-control-tools/patch-recipes/update-chainparams.sh
+++ b/contrib/source-control-tools/patch-recipes/update-chainparams.sh
@@ -23,13 +23,18 @@ while [ "${INDEX}" -lt "${#ARGS[@]}" ]; do
   ((INDEX+=1))
   TESTNET_PORT="${ARGS[${INDEX}]}"
   ((INDEX+=1))
+  if [[ -z "$CONFIG_FILE" ]]; then
+    CONFIG_FLAG="--config=~/.lotus/lotus.conf"
+  else
+    CONFIG_FLAG="--config=${CONFIG_FILE}"
+  fi
 
   CHAINPARAMS_MAINNET_TXT="${NETWORK}/chainparams_main.txt"
-  ./make_chainparams.py -a 127.0.0.1:"${MAINNET_PORT}"> "${CHAINPARAMS_MAINNET_TXT}"
+  ./make_chainparams.py -a 127.0.0.1:"${MAINNET_PORT}" "${CONFIG_FLAG}" > "${CHAINPARAMS_MAINNET_TXT}"
   git add "${CHAINPARAMS_MAINNET_TXT}"
 
   CHAINPARAMS_TESTNET_TXT="${NETWORK}/chainparams_test.txt"
-  ./make_chainparams.py -a 127.0.0.1:"${TESTNET_PORT}" > "${CHAINPARAMS_TESTNET_TXT}"
+  ./make_chainparams.py -a 127.0.0.1:"${TESTNET_PORT}" "${CONFIG_FLAG}" > "${CHAINPARAMS_TESTNET_TXT}"
   git add "${CHAINPARAMS_TESTNET_TXT}"
 
   CHAINPARAMS_CONSTANTS="${TOPLEVEL}"/src/networks/"${NETWORK}"/chainparamsconstants.cpp
